### PR TITLE
[Tank] Split mitigation checks into physical and magical damage

### PR DESCRIPTION
**Pull request by hpabst**
This pull request changes the existing mitigation check module to differentiate between physical and magical damage type checks.

### CODE REVIEW

File: src/parser/paladin/protection/modules/features/ShieldOfTheRighteous.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ class ShieldOfTheRighteous extends Analyzer {
   constructor(...args) {
     super(...args);
     const boss = findByBossId(this.owner.boss.id);
-    this._tankbusters = boss.fight.softMitigationChecks;
+    this._tankbusters = [...boss.fight.softMitigationChecks.physical, ...boss.fight.softMitigationChecks.magical];
```
Comment (emallson) [inline src/parser/paladin/protection/modules/features/ShieldOfTheRighteous.js]: this should only have physical. SotR does nothing against magic damage as of 8.0

File: src/parser/shared/modules/MitigationCheck.js [hunk 2]
```diff
@@ -19,47 +20,77 @@ class MitigationCheck extends Analyzer {
...
+      debuffCheck = [...this.debuffCheckPhysical, ...this.debuffCheckPhysAndMag];
+    } else {
+      checks = this.checksMagical;
+      buffCheck = [...this.buffCheckMagical, ...this.buffCheckPhysAndMag];
+      debuffCheck = [...this.debuffCheckMagical, ...this.debuffCheckPhysAndMag];
+    }
+    if (checks.includes(spell) && !event.tick) {
+      debug && console.log(buffCheck);
+      debug && console.log(debuffCheck);
+      if (buffCheck.some((e) => this.selectedCombatant.hasBuff(e)) || event.hitType === HIT_TYPES.IMMUNE) {
```
Comment (emallson) [inline src/parser/shared/modules/MitigationCheck.js]: this function got a bit messy in the transition. idk if there is a great way to simplify it, though

File: src/parser/paladin/protection/modules/features/ShieldOfTheRighteous.js (hunk 1 above)
Comment (hpabst) [reply src/parser/paladin/protection/modules/features/ShieldOfTheRighteous.js]: Fixed.

File: src/parser/shared/modules/MitigationCheck.js (hunk 2 above)
Comment (hpabst) [reply src/parser/shared/modules/MitigationCheck.js]: My own feeling is that it's fine for the time being. It's a little bit ugly, but like you said there's no obvious way to simplify things.

**Pull request merged**